src/main/scala/nl.knaw.dans.easy.deposit/StateManager.scala

```diff
@@ -0,0 +1,147 @@
+          case "REJECTED" => StateInfo(State.rejected, getStateDescription(submittedProps))
+          case "FEDORA_ARCHIVED" => StateInfo(State.archived, s"The dataset is published at ${ landingPage(tryDoi = true) }")
+          case "FAILED" => StateInfo(State.inProgress, s"The deposit is in progress.")
+          case "IN_REVIEW" => StateInfo(State.inProgress, s"The deposit is available at ${ landingPage(tryDoi = false) }")
+          case str: String => throw InvalidPropertyException(stateLabelKey, str, submittedProps)
+        }
+        saveNewState(newState)
+        newState
+      case draftState @ (State.draft | State.rejected | State.archived) =>
+        StateInfo(draftState, getStateDescription(draftProps))
+    }
+  }
+
+  def canChangeState(newStateInfo: StateInfo): Try[Unit] = getStateInfo.flatMap { oldStateInfo =>
+    // changeState returns the same exception but the submitter should not start without checking
+    val oldState = oldStateInfo.state
+    val newState = newStateInfo.state
+    (oldState, newState) match {
+      case (State.draft, State.submitted) |
+           (State.rejected, State.draft) => Success(())
+      case _ => Failure(IllegalStateTransitionException(oldState, newState))
+    }
+  }
+
+  def changeState(newStateInfo: StateInfo): Try[Unit] = getStateInfo.flatMap { old =>
+    // getStateInfo has been called by canChangeState, but it is not an IO action
+    // so let's keep it simple without optimisation
+    (old.state, newStateInfo.state) match {
+      case (State.draft, State.submitted) =>
+        val bagStoreBagId = UUID.randomUUID()
+        // probably properly saved without toString but getSubmittedBagId would throw
+        // ConversionException: 'bag-store.bag-id' doesn't map to a String object
+        draftProps.setProperty(bagIdKey, bagStoreBagId.toString)
+        saveNewState(newStateInfo)
+        Success(())
+      case (State.rejected, State.draft) =>
+        draftProps.clearProperty(bagIdKey)
+        saveNewState(newStateInfo)
+        Success(())
+      case (oldState, newState) =>
+        Failure(IllegalStateTransitionException(oldState, newState))
+    }
+  }
+
+  /** @return the value of bag-store.bag-id which is
+   *          - created by changeState when changed to SUBMITTED
+   *          - destroyed by changeState when changed to DRAFT
+   *          fails with:
+   *          PropertyException (if bag-store.bag-id is not found)
+   *          ConversionException (if setProperty was called with something else than a String)
+   *          IllegalArgumentException (if the value is not a valid UUID)
+   */
+  def getSubmittedBagId: Try[UUID] = Try {
+    // exception intercepted as not expected
+    // by TriedActionResult.getOrRecoverWithActionResult
+    UUID.fromString(draftProps.getString(bagIdKey))
+  }
+
+  val baseURL = "https://easy.dans.knaw.nl/ui"
+
+  private def landingPage(tryDoi: Boolean) = {
+    Try { getProp("identifier.fedora", submittedProps) }
```

Comment:
So far never implemented.

Reply:
To the best of my knowledge, it is `easy-ingest-flow` that sets this property after it received the `easy-dataset:XXX` identifier from Fedora.